atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -132,8 +134,9 @@ public static CassandraService createInitialized(
     public void close() {}
 
     public ImmutableSet<CassandraServer> refreshTokenRangesAndGetServers() {
-        ImmutableSet.Builder<CassandraServer> servers = ImmutableSet.builder();
-        ImmutableMap.Builder<CassandraServer, String> hostToDatacentersThisRefresh = ImmutableMap.builder();
+        // explicitly not using immutable builders to deduplicate nodes
+        Set<CassandraServer> servers = new HashSet<>();
```

Comment:
You can continue to use a builder here - ImmutableSet explicitly ignores duplicate entries after the first (https://guava.dev/releases/19.0/api/docs/com/google/common/collect/ImmutableSet.Builder.html), whereas ImmutableMap, as noted, explicitly throws on build (https://guava.dev/releases/19.0/api/docs/com/google/common/collect/ImmutableMap.Builder.html)

Will not block on this.